src/data/fintype/fin.lean

```diff
@@ -45,4 +47,24 @@ begin
   { rintro ⟨i, hi, rfl⟩, simpa },
 end
 
+lemma card_filter_univ_succ' (p : (fin (n + 1)) → Prop) [decidable_pred p] :
+  (univ.filter p).card = (ite (p 0) 1 0) + (univ.filter (p ∘ fin.succ)).card :=
+begin
+  rw [fin.univ_succ, filter_cons, card_disj_union, map_filter, card_map],
+  split_ifs; simp,
+end
+
+lemma card_filter_univ_succ (p : (fin (n + 1)) → Prop) [decidable_pred p] : (univ.filter p).card =
+  if (p 0) then (univ.filter (p ∘ fin.succ)).card + 1 else (univ.filter (p ∘ fin.succ)).card :=
```

Comment:
Again, parens. Also, I think leaving a trailing `=`  on the previous line can make things harder to read:
```suggestion
lemma card_filter_univ_succ (p : fin (n + 1) → Prop) [decidable_pred p] :
  (univ.filter p).card = if p 0 then (univ.filter (p ∘ fin.succ)).card + 1
                                else (univ.filter (p ∘ fin.succ)).card :=
```